src/protobuf/buffer.cr

```diff
@@ -32,6 +32,29 @@ module Protobuf
       end
     end
 
+    def peek_uint64
+      n = shift = 0_u64
+      pos = @io.pos
+      loop do
+        if shift >= 64
+          raise Error.new("buffer overflow varint")
+        end
+        byte = @io.read_byte
+        if byte.nil?
+          @io.pos = pos
+          return nil
+        end
+        b = byte.unsafe_chr.ord
+
+        n |= ((b & 0x7F).to_u64 << shift)
+        shift += 7
+        if (b & 0x80) == 0
+          @io.pos = pos
+          return n.to_u64
+        end
+      end
+    end
```

Comment:
This method shares a lot of code with `read_uint64`. Do you know off-hand if it'd be possible to extract the duplicate code or implement one of them in terms of the other?

Reply:
As you've probably saw, I've refactored this by passing in a `peek` boolean.  Let me know if this isn't idiomatic Crystal code.